### PR TITLE
feature: aether.consumer library

### PR DESCRIPTION
**Pull request by shawnsarwar**
This library will be the basis for future Kafka consumers written in python for Aether. It makes is much simpler to deserialize messages from Avro into JSON based on their schema. It also introduced new features to the KafkaConsumer to handle some Aether specific syntax which are passed as metadata in Avro schemas.

Newly supported via syntax and aether.consumer library:

- Masking (removal) of fields from published messages via classification in schema/ subscription to a maximum classification level
- Publish flag in individual messages causing non matching messages to be ignored by consumers.

### CODE REVIEW

File: aether-utils/aether-consumer/conf/pip/primary-requirements.py2.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#################################################################################################################
+
+jsonpath_ng
+kafka
+flake8
+mock
+pytest
+pytest-runner
+spavro
+
```
Comment (obdulia-losantos) [inline aether-utils/aether-consumer/conf/pip/primary-requirements.py2.txt]: I always say that alphabetic order is a thing, it was hard to detect that the missing package is `request` (comparing with python 3 version) :sweat_smile:

File: aether-utils/aether-consumer/conf/pip/requirements.py3.txt [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#########################################################################################
+#                                                                              		    #
+#                              Frozen requirements                             		    #
+#                                                                              		    #
+#                               DO NOT CHANGE !!!                              		    #
+#                                                                              		    #
+# To update this first update 'primary-requirements.txt'                       		    #
+# then run 'docker-compose run consumer-sdk{-py2}-test pip_freeze'	                    #
```
Comment (obdulia-losantos) [inline aether-utils/aether-consumer/conf/pip/requirements.py3.txt:8]: `-py2` ? same header for both environments? (just a question)

File: aether-utils/aether-consumer/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
```
Comment (obdulia-losantos) [inline aether-utils/aether-consumer/entrypoint.sh]: I expected `entrypoint_py3.sh` :joy:

**Review by obdulia-losantos — COMMENTED** [review]
Looks like it's all about... tests?

**Comment by shawnsarwar** [comment]
@obdulia-losantos Aligned the naming {-py2} and now {-py3} for parallel assets. Ordered dependencies, made changes according to other comments.

File: aether-utils/aether-consumer/conf/docker/setup.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
```
Comment (obdulia-losantos) [inline aether-utils/aether-consumer/conf/docker/setup.sh:1]: something in the air tells me that this file is useless and also not in use :wink:

File: aether-utils/aether-consumer/conf/pip/requirements.py3.txt (hunk 2 above)
Comment (obdulia-losantos) [reply aether-utils/aether-consumer/conf/pip/requirements.py3.txt:8]: I saw that you change the common header but not this one... neither the others. Anyway, it's going to be changed the next time someone upgrades the dependencies.

File: aether-utils/aether-consumer/entrypoint_py2.sh [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    ;;
+
+    pip_freeze )
+
+        rm -rf /tmp/env
+        pip install -f ./conf/pip/dependencies -r ./conf/pip/primary-requirements.py2.txt --upgrade
+
+        cat /code/conf/pip/requirements_header.txt | tee conf/pip/requirements.py2.txt
+        pip freeze --local | grep -v appdir | tee -a conf/pip/requirements.py2.txt
+    ;;
```
Comment (obdulia-losantos) [inline aether-utils/aether-consumer/entrypoint_py2.sh:59]: If you want to fix and make this work properly, removing the obsolete entries, the fix is this:

```bash
pip install virtualenv
rm -rf /tmp/env
virtualenv -p python3 /tmp/env/

/tmp/env/bin/pip install -f ./conf/pip/dependencies -r ./conf/pip/primary-requirements.py2.txt --upgrade

cat /code/conf/pip/requirements_header.txt | tee conf/pip/requirements.py2.txt
/tmp/env/bin/pip freeze | tee -a conf/pip/requirements.py2.txt
```

:thinking: :crossed_fingers:

File: aether-utils/aether-consumer/aether/consumer.py [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+                    if package_result.get("schema") or len(package_result["messages"]) > 0:
+                        partition_result.append(package_result)
+                if len(partition_result) > 0:
+                    name = "topic:%s-partition:%s" % (part.topic, part.partition)
+                    result[name] = partition_result
+        return result
+
+    def seek_to_beginning(self):
+        # We override this method to allow for seeking before any messages have been consumed
+        # as poll consumes a message. Since we're going to change offset, we don't care.
```
Comment (obdulia-losantos) [inline aether-utils/aether-consumer/aether/consumer.py:159]: :joy:

**Review by obdulia-losantos — APPROVED** [review]
OMG! There are more comments and tests than actual code :scream:

**Pull request merged**